pkg/controllers/sgx/controller.go

```diff
@@ -142,6 +147,11 @@ func (c *controller) UpdateDaemonSet(rawObj client.Object, ds *apps.DaemonSet) (
 		updated = true
 	}
 
+	if ds.Spec.Template.Spec.InitContainers[0].Image != dp.Spec.InitImage {
+		ds.Spec.Template.Spec.InitContainers[0].Image = dp.Spec.InitImage
+		updated = true
+	}
+
```

Comment:
we need to check  `dp.Spec.InitImage == ""`.  GPU  controller logic can be followed here. let's have this fix as a separate commit